lib/features/artifacts.js

```diff
@@ -142,7 +153,9 @@ export default class Artifacts {
       var res = yield waitForEvent(putReq, 'response');
```

Comment:
hmm... Maybe we need to refactor all of this a bit... Into something with abstractions... maybe:
`getFileStreamFromContainerFile(filePath, async (length, stream) => { /* upload logic */ }, , retries = 5)` and for directories:
`getFileStreamsFromContainerFolder(folderPath, async (length, stream) => { /* upload logic (called once for each task) */ }, retries = 5)`.

So that `getFileStreamFromContainerFile` handles get file... and tarextract etc... plus retries if needed... In folder it has to track which files it already uploaded successfully and just do `.drain()` for those on next try... well maybe.. I can see the best abstractions right now...